### PR TITLE
Add page titles for pages accessible by participant role

### PR DESCRIPTION
**Pull request by indigolain**
I've added page title for the pages accessible by participant role, and added base implementation for using the i18n feature in the process.

When specified, the page's title shows up like the screenshot below. If not specified, the page title will remain the original, which is `ConferenceApp`.

Screenshot 2024-08-18 at 19 11 04

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ module ConferenceApp
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 7.0
+    config.middleware.use Rack::Locale
```
Comment (indigolain) [inline config/application.rb:25]: `Rack::Locale` is used to determine which locale the application should use.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>ConferenceApp</title>
+    <title><%= page_title(yield(:page_title)) %></title>
```
Comment (indigolain) [inline app/views/layouts/application.html.erb:4]: Calling `content_for` for page_title in individual page configures the page title like https://github.com/kaigionrails/conference-app/pull/256/files#diff-d355e12f3e1517389297b283028f8ef3dde4e9d51127d3fbb8f57924b8770f29R1

File: Gemfile.lock [hunk 3]
```diff
@@ -283,31 +278,33 @@ GEM
...
-      actionview (= 7.1.3.4)
-      activejob (= 7.1.3.4)
-      activemodel (= 7.1.3.4)
-      activerecord (= 7.1.3.4)
-      activestorage (= 7.1.3.4)
-      activesupport (= 7.1.3.4)
+    rackup (1.0.0)
+      rack (< 3)
+      webrick
+    rails (7.2.0)
```
Comment (okuramasafumi) [inline Gemfile.lock:294]: Why upgrading Rails?

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -11,6 +11,11 @@ def application_url
     Rails.configuration.application_url
   end
 
+  def page_title(page_title = '')
```
Comment (okuramasafumi) [inline app/helpers/application_helper.rb:14]: I don't like the parameter name same as the method name, so maybe:

```suggestion
  def page_title(title = '')
```

File: sig/prototype/app/helpers/application_helper.rbs [hunk 5]
```diff
@@ -9,5 +9,7 @@ module ApplicationHelper
 
   def application_url: () -> untyped
 
+  def page_title: (?::String page_title) -> untyped
```
Comment (okuramasafumi) [inline sig/prototype/app/helpers/application_helper.rbs:12]:
```suggestion
  def page_title: (?::String page_title) -> string
```

File: Gemfile.lock (hunk 3 above)
Comment (indigolain) [reply Gemfile.lock]: Sorry, I have no idea why I did that 🙏 
Returned the Rails version with https://github.com/kaigionrails/conference-app/pull/256/commits/5da601ca61bd80943689286a9b582084e765d9d4

File: sig/prototype/app/helpers/application_helper.rbs (hunk 5 above)
Comment (indigolain) [reply sig/prototype/app/helpers/application_helper.rbs]: Thank you! Changed it to `string` with https://github.com/kaigionrails/conference-app/pull/256/commits/b595e690f7a46cd3d977995e3f1845cdb9cab6d7

File: app/helpers/application_helper.rb (hunk 4 above)
Comment (indigolain) [reply app/helpers/application_helper.rb]: Renamed with https://github.com/kaigionrails/conference-app/pull/256/commits/2f81f47d6870139d3fd6081c376285d9d54748a6

**Comment by okuramasafumi** [comment]
@indigolain Thank you for the fix, could you follow standardrb rules? (Simply running `bundle exec standardrb --fix` will work)

**Comment by indigolain** [comment]
@okuramasafumi 
Applied the lint via `bundle exec standardrb --fix` with https://github.com/kaigionrails/conference-app/pull/256/commits/912c6ab439ed6d893fa658c198eff50ba3b3c1de!

**Review by unasuke — APPROVED** [review]
Thankyou!

**Pull request merged**